testsuite/cluster-test/src/experiments/twin_validator.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::{
+    cluster::Cluster,
+    experiments::{Context, Experiment, ExperimentParam},
+    instance,
+    instance::Instance,
+    tx_emitter::EmitJobRequest,
+    util::unix_timestamp_now,
+};
+use async_trait::async_trait;
+use futures::future::try_join_all;
+use libra_logger::info;
+use rand::Rng;
+use std::{collections::HashSet, fmt, time::Duration};
+use structopt::StructOpt;
+use tokio::time;
+
+#[derive(StructOpt, Debug)]
+pub struct TwinValidatorsParams {
+    #[structopt(long, default_value = "1", help = "Set twin node pair number")]
+    pub pair: usize,
+}
+
+pub struct TwinValidators {
+    instances: Vec<Instance>,
+    twin_validators: Vec<Instance>,
+}
+
+impl ExperimentParam for TwinValidatorsParams {
+    type E = TwinValidators;
+    fn build(self, cluster: &Cluster) -> Self::E {
+        if self.pair > 3 {
+            panic!("pair number {} can not more than 9", self.pair);
+        }
+        let mut instances = cluster.validator_instances().to_vec();
+        let mut twin_validators = vec![];
+        let mut rnd = rand::thread_rng();
+        for _i in 0..self.pair {
+            twin_validators.push(instances.remove(rnd.gen_range(0, instances.len())));
+        }
+        Self::E {
+            instances,
+            twin_validators,
+        }
+    }
+}
+
+#[async_trait]
+impl Experiment for TwinValidators {
+    fn affected_validators(&self) -> HashSet<String> {
+        instance::instancelist_to_set(&self.twin_validators)
+    }
+
+    async fn run(&mut self, context: &mut Context<'_>) -> anyhow::Result<()> {
+        let buffer = Duration::from_secs(60);
+        let window = Duration::from_secs(240);
+        let mut new_instances = vec![];
+        for i in 0..self.twin_validators.len() {
```

Comment:
With current way experiment is written there is one problem - we do not know if at some point spawning twin will just completely break (e.g. for example won't launch twin).

To avoid this I propose following:
* Stop original validator before starting twin
* After starting twin, do new_inst.wait_for_json_rpc to confirm it has started
* Start original validator (without waiting for jsonrpc)

Reply:
good to know this way to check ```do new_inst.wait_for_json_rpc to confirm it has started```